tracer/src/Datadog.Trace/ClrProfiler/InstrumentationDefinitions.cs

```diff
@@ -9,9 +9,23 @@ internal static partial class InstrumentationDefinitions
     {
         private static string assemblyFullName = typeof(InstrumentationDefinitions).Assembly.FullName;
 
-        internal static NativeCallTargetDefinition[] GetAllDefinitions()
+        internal static Payload GetAllDefinitions()
         {
-            return GetDefinitionsArray();
+            return new Payload
+            {
+                // Fixed Id for definitions payload (to avoid loading same integrations from multiple AppDomains)
+                DefinitionsId = "FFAFA5168C4F4718B40CA8788875C2DA",
```

Comment:
Will this ever change? Do you expect a scenario where we'd need to load definitions twice?

Reply:
I don't expect that scenario tbh. I think we should load the integrations only once (like we did with the json file). So I don't expect that Id to change.